agr_api/src/main/java/org/alliancegenome/api/service/ExpressionRibbonESService.java

```diff
@@ -120,8 +120,16 @@ private RibbonEntity getExpressionRibbonSummary(String geneID) {
 		Pagination pagination = new Pagination(1, 250000, null, null);
 		JsonResultResponse<GeneExpressionDocument> expressionAnnotations = expressionService.getExpressionAnnotations(List.of(geneID), null, null, pagination);
 
-		Gene gene = expressionAnnotations.getResults().get(0).getGeneExpressionAnnotation().getExpressionAnnotationSubject();
-		String dataProvider = expressionAnnotations.getResults().get(0).getGeneExpressionAnnotation().getDataProvider().getAbbreviation();
+		String dataProvider;
+		Gene gene;
+		if (CollectionUtils.isNotEmpty(expressionAnnotations.getResults())) {
+			gene = expressionAnnotations.getResults().get(0).getGeneExpressionAnnotation().getExpressionAnnotationSubject();
```

Comment:
I would define an annotation object used in line 126 and 127
GeneExpressionAnnotation annotation = expressionAnnotations.getResults().get(0).getGeneExpressionAnnotation();
the two assignments then get more readable.